src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -141927,5 +141927,652 @@ static T GetValue2<T>(Collection c, object key)
+
+#nullable enable
+public class C
+{
+    void M()
+    {
+        _ = I3.Inner.Item;
+        I3.Inner.Item = null;
+    }
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.NetStandardLatest);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void AmbigMember_DifferenceBetweenNonnullableAndOblivious_WithoutConstraint_Method()
+        {
+            var src = @"
+#nullable disable
+interface I<T> { ref T Get(); }
+
+#nullable enable
+interface I2<T> : I<T> { }
+
+#nullable disable
+interface I3 : I<object>, I2<object> { }
+
+#nullable enable
+public class C
+{
+    void M(I3 i)
+    {
+        _ = i.Get();
+        i.Get() = null;
+    }
+}
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics();
+
+            var tree = comp.SyntaxTrees.Single();
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
+            var item = tree.GetRoot().DescendantNodes().OfType<InvocationExpressionSyntax>().First();
+            Assert.Equal("i.Get()", item.ToString());
+            Assert.Equal("ref System.Object I<System.Object>.Get()", model.GetSymbolInfo(item).Symbol.ToTestDisplayString());
+        }
+
+        [Fact]
+        public void AmbigMember_DifferenceBetweenNonnullableAndOblivious_WithoutConstraint_ReverseOrder()
+        {
+            var src = @"
+#nullable disable
+interface I<T> { T Item { get; set; } }
+
+#nullable enable
+interface I2<T> : I<T> { }
+
+#nullable disable
+interface I3 : I2<object>, I<object> { }
+
+#nullable enable
+public class C
+{
+    void M(I3 i)
+    {
+        _ = i.Item;
+        i.Item = null;
+    }
+}
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics();
+            var i3 = comp.GetTypeByMetadataName("I3");
+            AssertEx.Equal(new string[] { "I2<object>", "I<object>" },
+                i3.Interfaces().ToTestDisplayStrings(TypeWithAnnotations.TestDisplayFormat));
+            AssertEx.Equal(new string[] { "I2<object>", "I<object>" },
+                i3.AllInterfaces().ToTestDisplayStrings(TypeWithAnnotations.TestDisplayFormat));
+
+            var tree = comp.SyntaxTrees.Single();
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
+            var item = tree.GetRoot().DescendantNodes().OfType<MemberAccessExpressionSyntax>().First();
+            Assert.Equal("i.Item", item.ToString());
+            Assert.Equal("System.Object I<System.Object>.Item { get; set; }", model.GetSymbolInfo(item).Symbol.ToTestDisplayString());
+        }
+
+        [Fact]
+        public void AmbigMember_DifferenceBetweenNonnullableAndOblivious_WithConstraint()
+        {
+            var src = @"
+#nullable disable
+interface I<T> where T : class { T Item { get; set; } }
+
+#nullable enable
+interface I2<T> : I<T> where T : class { }
+
+#nullable disable
+interface I3 : I<object>, I2<object> { }
```

Comment:
>I<object>, I2<object> [](start = 15, length = 21)

Does flipping the order here change what symbols we get?

Reply:
It does. See `AmbigMember_DifferenceBetweenNonnullableAndOblivious_WithConstraint_ReverseOrder`

---
In reply to: [544374723](https://github.com/dotnet/roslyn/pull/49338#discussion_r544374723) [](ancestors = 544374723)

Reply:
>It does. See AmbigMember_DifferenceBetweenNonnullableAndOblivious_WithConstraint_ReverseOrder

It doesn't look like we assert symbol in that test.

---
In reply to: [544603988](https://github.com/dotnet/roslyn/pull/49338#discussion_r544603988) [](ancestors = 544603988,544374723)